airbyte-server/src/main/java/io/airbyte/server/RunMigration.java

```diff
@@ -52,11 +53,12 @@
   public RunMigration(JobPersistence jobPersistence,
                       ConfigRepository configRepository,
                       String targetVersion,
-                      ConfigPersistence seedPersistence) {
+                      ConfigPersistence seedPersistence,
+                      SpecFetcher specFetcher) {
```

Comment:
is injecting the `SpecFetcher` here safe? While the migration is running, I think there is no guarantee that we can launch jobs in temporal. I know we block the scheduler starting up during a migration. I guess we technically don't block temporal?

Reply:
Needing to have access to a spec without having to run a docker container seems to be a recurring (stupid) problem. Wondering if we should just be storing the spec as past of source definition. Obviously out of scope for what you're doing, but what do you think we should be doing here overall?

If this is safe for now, then we should move forward with it. If it's not, then we should figure something else out 😄 . Either way, I think this might bite us in the future even if it is safe now, so it might be good to start to define what we truly want here.

Reply:
The scheduler/worker wait until the server is live before serving work orders from the Temporal service.

The `SpecFetcher` uses a "scheduler" client that actually doesn't interact with the scheduler at all; it just interacts with tracking (Segment) and the temporal service. No db tables are touched, no actual `Job` is created/scheduled.

I'm wondering if we can just remove the wait on the server from the worker. Then it could service these requests on demand. Maybe we should also rename `DefaultSynchronousSchedulerClient`...